frontend/src/component/segments/EditSegment/EditSegment.tsx

```diff
@@ -135,7 +138,8 @@ export const EditSegment = ({ modal }: IEditSegmentProps) => {
                 mode='edit'
             >
                 <UpdateButton
-                    permission={UPDATE_SEGMENT}
+                    permission={[UPDATE_SEGMENT, UPDATE_PROJECT_SEGMENT]}
+                    projectId={projectId}
```

Comment:
These 2 lines fix the bug on the UI, where the button was shown as being disabled.